.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,116 @@
+name: Benchmark
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-binary:
+    runs-on: self-hosted
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.tag }}
+      - name: Setup Rust
+        run: rustup show
+      - name: Build
+        run: |
+          cargo build --release --features runtime-benchmarks,turing-node
+          mkdir -p artifacts/
+          cp target/release/oak-collator artifacts/
+      - name: Upload binary
+        uses: actions/upload-artifact@v2
+        with:
+          name: oak-collator
+          path: artifacts/
+          retention-days: 1
+  run-benchmarks:
+    name: Run Benchmarks
+    runs-on: self-hosted
+    needs: build-binary
+    strategy:
+      matrix:
+        pallet:
+          - automation_time
+          - vesting
+          - valve
+          - xcmp_handler
+    steps:
+      - name: Download binary
+        uses: actions/download-artifact@v2
+        with:
+          name: oak-collator
+      - name: Execute pallet benchmark
+        run: |
+          chmod +x oak-collator
+          ./oak-collator \
+            benchmark \
+            pallet \
+            --chain turing-dev \
+            --execution wasm \
+            --wasm-execution compiled \
+            --pallet pallet_${{ matrix.pallet }} \
+            --extrinsic '*' \
+            --repeat 20 \
+            --steps 50 \
+            --output ./${{ matrix.pallet }}-raw-weights.rs \
+            --template ./.maintain/frame-weight-template.hbs
+      - name: Upload raw weights
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.pallet }}
+          path: ${{ matrix.pallet }}-raw-weights.rs
```

Comment:
Any reason to have retention policy here as well?

Reply:
The default of 90 days seemed reasonable.  The only reason the `oak-collator` has one is that I don't consider it an artifact of the job. I just abuse artifacts to pass a specific version of the binary to a later job for use.